src/blocks/scratch3_pen.js

```diff
@@ -171,44 +184,22 @@ class Scratch3PenBlocks {
     }
 
     /**
-     * Update the cached color from the hue, shade and transparency values in the provided
-     * PenState object.
-     * @param {PenState} penState - the pen state to update.
-     * @private
-     */
-    _updatePenColor (penState) {
-        let rgb = Color.hsvToRgb({h: penState.hue * 180 / 100, s: 1, v: 1});
-        const shade = (penState.shade > 100) ? 200 - penState.shade : penState.shade;
-        if (shade < 50) {
-            rgb = Color.mixRgb(Color.RGB_BLACK, rgb, (10 + shade) / 60);
-        } else {
-            rgb = Color.mixRgb(rgb, Color.RGB_WHITE, (shade - 50) / 60);
-        }
-        penState.penAttributes.color4f[0] = rgb.r / 255.0;
-        penState.penAttributes.color4f[1] = rgb.g / 255.0;
-        penState.penAttributes.color4f[2] = rgb.b / 255.0;
-        penState.penAttributes.color4f[3] = this._transparencyToAlpha(penState.transparency);
-    }
-
-    /**
-     * Wrap a pen hue or shade values to the range (0,200).
-     * @param {number} value - the pen hue or shade value to the proper range.
+     * Wrap a color input into the range (0,100).
+     * @param {number} value - the value to be wrapped.
      * @returns {number} the wrapped value.
      * @private
      */
-    _wrapHueOrShade (value) {
-        value = value % 200;
-        if (value < 0) value += 200;
-        return value;
+    _wrapColor (value) {
+        return MathUtil.wrapClamp(value, 0, 100);
     }
 
     /**
-     * Clamp a pen transparency value to the range (0,100).
-     * @param {number} value - the pen transparency value to be clamped.
+     * Clamp a pen color parameter to the range (0,100).
+     * @param {number} value - the value to be clamped.
      * @returns {number} the clamped value.
      * @private
      */
-    _clampTransparency (value) {
+    _clampColorParam (value) {
```

Comment:
prio-low: the names `_wrapColorInput` and `_clampColorParam` are very similar and don't immediately communicate the difference between the two, other than wrap vs. clamp. If that's the only intended difference then I would prefer consistency between "input" and "param" in these names. I don't think this needs to be fixed immediately.

Reply:
Sorry for merging this before getting your comments!

re: compatibility, I have on my list (but not in an issue yet) that we will need to update the sb2 importer and add back in the scratch 2.0 implementations (including different wrapping and clamping behavior, the shade block, and accepting hex values). Then, once we have auto-loading of then pen extension for projects containing pen blocks, they should work.

I'm not totally sure what to do about the wrap and clamp function names. We will need at least two more variants of wrapping and clamping to different ranges when we implement the 2.0 style pen color blocks, so maybe we can think about the naming scheme again then. The use of "color" as one of the color parameters complicates things- I was thinking about switching to using "hue" in the code, but I tried to just keep it simple and match what the user   sees.